crates/proof-of-sql-planner/src/plan.rs

```diff
@@ -0,0 +1,472 @@
+    let input_schema = schemas
+        .get(table_name)
+        .ok_or_else(|| PlannerError::TableNotFound {
+            table_name: table_name.to_string(),
+        })?;
+    // Get aliased expressions
+    let aliased_dyn_proof_exprs =
+        get_aliased_dyn_proof_exprs(&table_ref, projection, input_schema, projected_schema)?;
+    let column_fields = aliased_dyn_proof_exprs
+        .iter()
+        .map(|aliased_expr| {
+            ColumnField::new(aliased_expr.alias.clone(), aliased_expr.expr.data_type())
+        })
+        .collect::<Vec<_>>();
+    let table_exec = DynProofPlan::new_table(table_ref, column_fields);
+    Ok(DynProofPlan::new_projection(
+        aliased_dyn_proof_exprs,
+        table_exec,
+    ))
+}
+
+/// Convert a `TableScan` with filters but without fetch limit to a `DynProofPlan`
+fn table_scan_to_filter(
+    table_name: &TableReference,
+    schemas: &IndexMap<TableReference, DFSchema>,
+    projection: Option<Vec<usize>>,
+    projected_schema: &DFSchema,
+    filters: &[Expr],
+) -> PlannerResult<DynProofPlan> {
+    // Check if the table exists
+    let table_ref = table_reference_to_table_ref(table_name)?;
+    let table_expr = TableExpr {
+        table_ref: table_ref.clone(),
+    };
+    let input_schema = schemas
+        .get(table_name)
+        .ok_or_else(|| PlannerError::TableNotFound {
+            table_name: table_name.to_string(),
+        })?;
+    // Get aliased expressions
+    let aliased_dyn_proof_exprs =
+        get_aliased_dyn_proof_exprs(&table_ref, projection, input_schema, projected_schema)?;
+    // Process filter
+    let filter_proof_exprs = filters
+        .iter()
+        .map(|f| expr_to_proof_expr(f, input_schema))
+        .collect::<PlannerResult<Vec<_>>>()?;
+    // Filter
+    let consolidated_filter_proof_expr = filter_proof_exprs[1..]
+        .iter()
+        .cloned()
+        .try_fold(filter_proof_exprs[0].clone(), DynProofExpr::try_new_and)?;
+    Ok(DynProofPlan::new_filter(
+        aliased_dyn_proof_exprs,
+        table_expr,
+        consolidated_filter_proof_expr,
+    ))
+}
+
+/// Visit a [`datafusion::logical_plan::LogicalPlan`] and return a [`DynProofPlan`]
+pub fn logical_plan_to_proof_plan(
+    plan: &LogicalPlan,
+    schemas: &IndexMap<TableReference, DFSchema>,
+) -> PlannerResult<DynProofPlan> {
+    match plan {
+        LogicalPlan::EmptyRelation { .. } => Ok(DynProofPlan::new_empty()),
+        // No filter or fetch limit
+        LogicalPlan::TableScan(TableScan {
+            table_name,
+            projection,
+            projected_schema,
+            filters,
+            fetch: None,
+            ..
+        }) if filters.is_empty() => {
+            table_scan_to_projection(table_name, schemas, projection.clone(), projected_schema)
+        }
+        // Filter but no fetch limit
+        LogicalPlan::TableScan(TableScan {
+            table_name,
+            projection,
```

Comment:
Can we assume this if the LogicalPlan is resolved?
```suggestion
            projection: Some(proj),
```
If we won't ever hit the `None` case in production, I would rather not have code handling it. (Or add it later.)